lib/util/device-service-details.ts

```diff
@@ -126,24 +126,30 @@ export const generateCurrentServiceDetails = (
 	const device = (rawDevice as Device) as DeviceWithServiceDetails;
 
 	// Essentially a groupBy(installs, 'service_name')
-	// but try making it a bit faster for the sake of large fleets
-	const currentServicesGroupedByName: Record<string, CurrentService[]> = {};
-	installs.forEach(container => {
-		let serviceContainerGroup =
-			currentServicesGroupedByName[container.service_name];
-		if (!serviceContainerGroup) {
-			serviceContainerGroup = currentServicesGroupedByName[
-				container.service_name
-			] = [];
+	// but try making it a bit faster for the sake of large fleets.
+	// Uses Object.create(null) so that there are no inherited properties
+	// which could match service names
+	const currentServicesGroupedByName: Record<
+		string,
+		CurrentService[]
+	> = Object.create(null);
+	for (const container of installs) {
```

Comment:
This can be
```suggestion
	for (const { service_name } of installs) {
```

Reply:
Can't do @Page- , since we also need the whole `container` at the end.
PS: You had me for more than a moment, I almost typed this change before realizing... :)